pkg/apis/pingcap/v1alpha1/tidbmonitor_types.go

```diff
@@ -108,6 +117,9 @@ type PrometheusSpec struct {
 
 	// +optional
 	Config *PrometheusConfiguration `json:"config,omitempty"`
+
+	// Disable prometheus compaction.
+	DisableCompaction bool `json:"disableCompaction,omitempty"`
```

Comment:
How about always disable compaction if `monitor.Spec.Thanos != nil` as you did before?

Reply:
Done